res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1600 @@
+	  const deck = DDJSR2.channelGroups[group];
+        const activate = value > 0;
+        if (activate) { // act on button press
+            engine.softStart(deck+1, true);
+        }
+    };
+
+    this.jogRingTick = function(channel, control, value, status, group) {
+        DDJSR2.pitchBendFromJog(group, DDJSR2.getJogWheelDelta(value));
+    };
+
+    this.jogRingTickShift = function(channel, control, value, status, group) {
+	  DDJSR2.pitchBendFromJog(
+            group,
+            DDJSR2.getJogWheelDelta(value) * DDJSR2.jogwheelShiftMultiplier
+        );
+    };
+
+    this.vinylPlatterTick = function(channel, control, value, status, group) {
+	  const deck = DDJSR2.channelGroups[group];
+
+        if (DDJSR2.scratchMode[deck] && engine.isScratching(deck + 1)) {
+	      engine.scratchTick(deck + 1, DDJSR2.getJogWheelDelta(value));
+	  }
+    };
+
+    this.jogPlatterTick = function(channel, control, value, status, group) {
+        const deck = DDJSR2.channelGroups[group];
+	  DDJSR2.pitchBendFromJog(group, DDJSR2.getJogWheelDelta(value));
+    };
+
+    this.jogPlatterTickShift = function(channel, control, value, status, group) {
+        const deck = DDJSR2.channelGroups[group];
+
+        if (DDJSR2.scratchMode[deck] && engine.isScratching(deck + 1)) {
+	      engine.scratchTick(deck + 1, DDJSR2.getJogWheelDelta(value));
+	  } else {
+	      DDJSR2.pitchBendFromJog(
+		  group,
+		  DDJSR2.getJogWheelDelta(value) * DDJSR2.jogwheelShiftMultiplier
+	      );
+        }
+    };
+
+    this.platterTouch = function(channel, control, value, status, group) {
+	  const deck = DDJSR2.channelGroups[group];
+
+	  if (DDJSR2.scratchMode[deck]) {
+            if (value) {
+                engine.scratchEnable(
+                    deck + 1,
+                    DDJSR2.scratchSettings.jogResolution,
+                    DDJSR2.scratchSettings.vinylSpeed,
+                    DDJSR2.scratchSettings.alpha,
+                    DDJSR2.scratchSettings.beta,
+                    true
+		  );
+            } else {
+                engine.scratchDisable(deck + 1, true);
+            }
+	  }
+    };
+
+    this.vinylButton = new components.Button({
+        midi: [0x90+offset, 0x17],
+        type: components.Button.prototype.types.toggle,
+        input: function(channel, control, value, status, group) {
+            var deck = DDJSR2.channelGroups[group],
+	          midiBaseAddress = 0x90,
+                channel = 0x17,
+                midiOut = 0x00;
+
+	      if (value) {
+	          DDJSR2.scratchMode[deck] = !DDJSR2.scratchMode[deck];
+	          midi.sendShortMsg(
+                    midiBaseAddress + offset,
+                    channel,
+                    DDJSR2.scratchMode[deck]
+		  );
+            }
+        }
+    });
+	  // Defaulting the scratchmodes.
+    midi.sendShortMsg(
+        0x90 + offset,
+        0x17,
+        DDJSR2.scratchMode[offset]
+    );
```

Comment:
don't do explicit `sendShortMsg`. Instead call trigger on the component that controls the LED (the mapping probably already does that automatically anyways if you have implemented `trigger` correctly).